Providers/Profile.swift

```diff
@@ -234,9 +234,9 @@ open class BrowserProfile: Profile {
         // since the DB handles will create new DBs under the new profile folder.
         let isNewProfile = !files.exists("")
 
-        // Set up our database handles.
-        self.loginsDB = BrowserDB(filename: "logins.db", secretKey: BrowserProfile.loginsKey, files: files)
-        self.db = BrowserDB(filename: "browser.db", files: files)
+        // Set up our database handles. A crash is intended if the schema cannot be set up successfully.
+        self.loginsDB = try! BrowserDB(filename: "logins.db", secretKey: BrowserProfile.loginsKey, schema: LoginsSchema(), files: files)
```

Comment:
Note that we've evolved into an undesirable situation: the logins DB is always opened and initialized, even if it's not needed. Indeed, so is the browser DB!

We started with a lazy var.

We then moved to a flag.

Now we have neither: because `BrowserDB` isn't in control of the lifecycle of the connection, it needs to make sure that the DB is set up before its initializer returns.

That's bad for risk (we always try to read and maybe write two DBs on each app foregrounding), and it's bad for perf (we write to two databases on first run).

This is why I suggested pushing down to SwiftData in some way: it would allow us to tie init behavior to connection management. We're confident in our connection management.

Consider whether you achieve that easily enough via a delegate pattern — _e.g._, a closure with a weak ref to the `Schema` that `BrowserDB` passes down to SwiftData, which runs it each time it opens a new connection.

Reply:
Ok, yeah. You've convinced me :-)

I'll see if I can rework this patch a bit to just let SwiftData run the schema preparation when it is first needed instead.